### PR TITLE
Added the Get API in Go for the DeliveryserviceServers endpoint

### PR DESCRIPTION
**Pull request by ajschmidt**
This is the first of 6 methods in the API.  More to come.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: lib/go-tc/deliveryservice_servers.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+// DeliveryServiceServer ...
+type DeliveryServiceServer struct {
+	Server          *int             `json:"server"`
+	DeliveryService *int             `json:"deliveryService"`
+	LastUpdated     *TimeNoMod       `json:"lastUpdated" db:"last_updated"`
+}
+
+
+type DssServer struct {
```
Comment (rob05c) [inline lib/go-tc/deliveryservice_servers.go]: Go names should have abbreviations consistently cased (all caps, or all lower). This should be `DSSServer`

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+func (dss TODeliveryServiceServer) GetKeys() (map[string]interface{}, bool) {
+	if dss.DeliveryService == nil {
+		return map[string]interface{}{"deliveryservice": 0}, false
+	}
+	if dss.Server == nil {
+		return map[string]interface{}{"server": 0}, false
+	}
+	keys := make(map[string]interface{})
+	ds_id := *dss.DeliveryService
+	server_id := *dss.Server
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go]: Go names shouldn't have underscores, should be `dsID`, `serverID`

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+	}
+	return "unknown"
+}
+
+func (dss *TODeliveryServiceServer) GetType() string {
+	return "deliveryserviceServers"
+}
+
+func (dss *TODeliveryServiceServer) SetKeys(keys map[string]interface{}) {
+	ds_id, _ := keys["deliveryservice"].(int) //this utilizes the non panicking type assertion, if the thrown away ok variable is false i will be the zero of the type, 0 here.
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go]: Should be `dsID`

Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go]: Should be `dsID`

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+	dss.DeliveryService = &ds_id
+
+	server_id, _ := keys["server"].(int) //this utilizes the non panicking type assertion, if the thrown away ok variable is false i will be the zero of the type, 0 here.
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go]: Should be `serverID`

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+	errs := validation.Errors{
+		"deliveryservice": validation.Validate(dss.DeliveryService, validation.Required),
+		"server":          validation.Validate(dss.Server, validation.Required),
+	}
+
+	return tovalidate.ToErrors(errs)
+}
+
+//The TODeliveryServiceServer implementation of the Creator interface
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go]: The first word of a function comment should begin with the name of the function, `Create`. Not doing so breaks Godoc.

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+			}
+			return err, eType
+		}
+		log.Errorf("received non pq error: %++v from create execution", err)
+		return tc.DBError, tc.SystemError
+	}
+	defer resultRows.Close()
+
+	var ds_id int
+	var server_id int
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go]: `dsID`, `serverID`

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+	}
+	if rowsAffected < 1 {
+		return errors.New("no parameter with that id found"), tc.DataMissingError
+	}
+	if rowsAffected > 1 {
+		return fmt.Errorf("this create affected too many rows: %d", rowsAffected), tc.SystemError
+	}
+
+	err = tx.Commit()
+	if err != nil {
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go]: Nitpick: can be combined to `if err := tx.Commit(); err != nil {`

**Comment by rob05c** [comment]
Looks good. Tested against a prod database, produces the same output as Perl

**Pull request merged**